flink-runtime/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -137,18 +153,26 @@ public final void start() {
 	 * @throws Exception indicating that the rpc endpoint could not be started. If an exception occurs,
 	 * then the rpc endpoint will automatically terminate.
 	 */
-	public void onStart() throws Exception {}
+	public final void doStart() throws Exception {
```

Comment:
Maybe let's rename this method into `internalCallOnStart` so that users see that this method is an internal method. It is a bit unfortunate that with Java's accessors we are not able to restrict the accessibility of this method to only the `AkkaRpcActor`.